src/client.ts

```diff
@@ -0,0 +1,79 @@
+import { iZerionAPI } from "./interface";
+import {
+  ChainData,
+  FungiblePositionsResponse,
+  GetChainsResponse,
+  GetPortfolioResponse,
+  PortfolioData,
+  PositionData,
+} from "./types";
+
+const BASE_URL = "https://api.zerion.io/v1";
+
+// Utility function to make API requests with error handling
+async function fetchFromZerion<T>(
+  endpoint: string,
+  apiKey: string,
+  env?: string
+): Promise<T> {
+  const headers = {
+    accept: "application/json",
+    authorization: `Basic ${apiKey}`,
+    // Optionally add the "X-Env" header for testnet or other environments
+    ...(env ? { "X-Env": env } : {}),
+  };
+
+  const response = await fetch(`${BASE_URL}${endpoint}`, { headers });
+
+  if (!response.ok) {
+    throw new Error(`Failed to fetch ${endpoint}: ${response.statusText}`);
+  }
+  const data = await response.json();
+  return data as T;
+}
+
+export class ZerionAPI implements iZerionAPI {
```

Comment:
Why does this class need an interface?

Reply:
no good reason. I could move all the doc strings from the interface into the implementation and delete this. It was really just a blueprint for me to get started.

Reply:
No more interface.